plugins/woocommerce-admin/client/marketing/overview-multichannel/Campaigns/CreateNewCampaignModal/CreateNewCampaignModal.tsx

```diff
@@ -0,0 +1,164 @@
+	const { className, ...restProps } = props;
+	const [ collapsed, setCollapsed ] = useState( true );
+	const { data: newCampaignTypes } = useNewCampaignTypes();
+	const { data: recommendedChannels } = useRecommendedChannels();
+	const { refetch } = useRegisteredChannels();
+
+	return (
+		<Modal
+			className={ classnames(
+				className,
+				'woocommerce-marketing-create-campaign-modal'
+			) }
+			title={ __( 'Create a new campaign', 'woocommerce' ) }
+			{ ...restProps }
+		>
+			<div className="woocommerce-marketing-new-campaigns">
+				<div className="woocommerce-marketing-new-campaigns__question-label">
+					{ __(
+						'Where would you like to promote your products?',
+						'woocommerce'
+					) }
+				</div>
+				<div>
+					{ newCampaignTypes?.map( ( el ) => {
+						return (
+							<Flex
+								key={ el.id }
+								className="woocommerce-marketing-new-campaign-type"
+								gap={ 4 }
+							>
+								<FlexItem>
+									<img
+										src={ el.icon }
+										alt={ el.name }
+										width="32"
+										height="32"
+									/>
+								</FlexItem>
+								<FlexBlock>
+									<Flex direction="column" gap={ 1 }>
+										<FlexItem className="woocommerce-marketing-new-campaign-type__name">
+											{ el.name }
+										</FlexItem>
+										<FlexItem className="woocommerce-marketing-new-campaign-type__description">
+											{ el.description }
+										</FlexItem>
+									</Flex>
+								</FlexBlock>
+								<FlexItem>
+									<Button
+										variant="secondary"
+										href={ el.createUrl }
+										target={
+											isExternalURL( el.createUrl )
+												? '_blank'
+												: '_self'
+										}
+									>
+										<Flex gap={ 1 }>
+											<FlexItem>
+												{ __(
+													'Create',
+													'woocommerce'
+												) }
+											</FlexItem>
+											{ isExternalURL( el.createUrl ) && (
+												<FlexItem>
+													<Icon
+														icon={ external }
+														size={ 16 }
+													/>
+												</FlexItem>
+											) }
+										</Flex>
+									</Button>
+								</FlexItem>
+							</Flex>
+						);
+					} ) }
+				</div>
+			</div>
+			{ !! recommendedChannels?.length && (
+				<div className="woocommerce-marketing-add-channels">
+					<Flex direction="column">
+						<FlexItem>
+							<Button
+								variant="link"
+								onClick={ () => setCollapsed( ! collapsed ) }
+							>
+								{ __(
+									'Add channels for other campaign types',
+									'woocommerce'
+								) }
+								<Icon
+									icon={ collapsed ? chevronDown : chevronUp }
+									size={ 24 }
+								/>
+							</Button>
+						</FlexItem>
+						{ ! collapsed && (
+							<FlexItem>
+								{ recommendedChannels.map( ( el, idx ) => {
+									return (
+										<Fragment key={ el.plugin }>
+											<SmartPluginCardBody
+												plugin={ el }
+												onInstalledAndActivated={
+													refetch
```

Comment:
Good catch! 

I made the fix in [`8b6aad4` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/8b6aad4138661e5a98386a67514adbba8e2bbeb7) for this CreateNewCampaignModal.

I also made the fix in [`6446053` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/644605378493bd649de8292292d9504ca9b6032a). When users activate a channel in the Channels card, we should also refetch the campaign types.

---

Food for thought / for future improvement: right now we are calling `refetchCampaignTypes` and `refetchRegisteredChannels` in an event handler in the above two places. I think it might be better if we come up with a wp.data action that contains these, e.g. `install channel -> activate channel -> invalidate resolution for campaign types and registered channels`.

Reply:
> I made the fix in [`8b6aad4` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/8b6aad4138661e5a98386a67514adbba8e2bbeb7) for this CreateNewCampaignModal.

After the 8b6aad4, it still won't refetch the data of `getCampaignTypes`.

The call to `invalidateResolution` in `useCampaignTypes` requires the second parameter:

```js
invalidateResolution( 'getCampaignTypes', [] );
```

Reply:
The function signature of `invalidateResolution` has been changed in WP version >= 6.0
- In WP version < 6.0, the second parameter has no default value
- In WP version >= 6.0, the second parameter defaults by `[]`

Therefore, with a WP version < 6.0, it cannot trigger state invalidation because of the mismatched `args` of a wp-data selector.

Ref:
- WP version < 6.0
   - https://github.com/WordPress/gutenberg/blob/%40wordpress/data%406.2.1/packages/data/src/redux-store/metadata/reducer.ts#L53-L57
- WP version >= 6.0
   - https://github.com/WordPress/gutenberg/blob/%40wordpress/data%406.3.0/packages/data/src/redux-store/metadata/reducer.ts#L56-L60
   - https://github.com/WordPress/gutenberg/blob/%40wordpress/data%406.3.0/packages/data/src/redux-store/metadata/utils.ts#L48-L51

Reply:
Got it. Similar to https://github.com/woocommerce/woocommerce/pull/37044#discussion_r1132055563, we need to make it work for WP 5.9, so I have made the fix in [`c6e6f87` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/c6e6f8774d6ec36f0d7ebe2c3e17e50d5c847e7c). I specified empty array dependency `[]` not just for the `invalidateResolution` but also for `hasFinishedResolution`, to make both of them consistent.